packages/expo-updates/ios/EXUpdates/AppLoader/FileDownloader.swift

```diff
@@ -805,10 +817,20 @@ public final class FileDownloader {
 
   private func downloadData(
     withRequest request: URLRequest,
+    progressBlock: FileDownloadProgressBlock?,
     successBlock: @escaping SuccessBlock,
     errorBlock: @escaping ErrorBlock
   ) {
-    let task = session.dataTask(with: request) { data, response, error in
+    var task: URLSessionTask?
+    task = session.dataTask(with: request) { data, response, error in
+      // when the task completes, invalidate and remove its observer
+      if let task = task {
```

Comment:
```suggestion
      if let task = task {
        task.progress.invalidate()
```

i think KVO is fine. not sure the intention to keep `progressObservations` map?

can we just register KVO and unregister it here without storing task<->KVO in the map?

Reply:
Thank you! It made it much [cleaner](https://github.com/expo/expo/pull/38307/commits/0b1f6682571882be77413ec3a4d74dde5acf7a56).